### ISSUE TITLE
XSS issue by parsing html codes

### ISSUE THREAD

**Issue by knowlet** [hackmdio/hackmd#379]
There seems to be a xss issue when parsing a block of html codes.
ex:
```
# Title
<h1>
a
</h1>
<img src=x onerror=alert(/Hello/)//
<script>alert(1);</script>
aaa
z
```

And the xss works in the editor and presentation mode.
https://hackmd.io/p/Sk25K469x#/

**Comment by jackycute** [comment]
Issue confirmed, thanks for reporting this.

**Comment by jackycute** [comment]
This is caused by the img `onerror` attribute and browser will auto close unmatched div tag.

**Comment by jackycute** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Fix XSS](https://trello.com/c/1xZuHSIL/3-fix-xss)

**Comment by jackycute** [comment]
Fixed in f491cdabc1dd82e93523ac0e156a669d9f37ba89
